### PR TITLE
Create "leshan-core-cf" project.

### PR DESCRIPTION
**Pull request by sbernard31**
There is now a new java project which contains common code between client and server which depends on Californium. (see #222)

### CODE REVIEW

File: leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ExchangeUtil.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import javax.security.auth.x500.X500Principal;
+
+import org.eclipse.californium.core.server.resources.CoapExchange;
+import org.eclipse.californium.scandium.auth.PreSharedKeyIdentity;
+import org.eclipse.californium.scandium.auth.RawPublicKeyIdentity;
+import org.eclipse.leshan.core.request.Identity;
+
+public class ExchangeUtil {
+
+    private ExchangeUtil() {
```
Comment (msangoi) [inline leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ExchangeUtil.java]: not needed

File: leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ResponseCodeUtil.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ *******************************************************************************/
+package org.eclipse.leshan.core.californium;
+
+import org.eclipse.californium.core.coap.CoAP;
+import org.eclipse.leshan.ResponseCode;
+import org.eclipse.leshan.util.Validate;
+
+public class ResponseCodeUtil {
+
+    private ResponseCodeUtil() {
```
Comment (msangoi) [inline leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ResponseCodeUtil.java]: not needed

File: leshan-core-cf/pom.xml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        <dependency>
+            <groupId>org.eclipse.californium</groupId>
+            <artifactId>californium-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.californium</groupId>
+            <artifactId>scandium</artifactId>
+        </dependency>
+
+        <!-- test dependencies -->
```
Comment (msangoi) [inline leshan-core-cf/pom.xml]: which tests? :)

**Comment by sbernard31** [comment]
Ok I made your changes. this is integrated in master now :)

**Pull request merged**